controllers/gcpmachine_controller.go

```diff
@@ -301,6 +308,40 @@ func (r *GCPMachineReconciler) getOrCreate(scope *scope.MachineScope, computeSvc
 	return instance, nil
 }
 
+func (r *GCPMachineReconciler) getAddresses(computeSvc *compute.Service, instance *gcompute.Instance) (*[]v1.NodeAddress, error) {
```

Comment:
Same here, seems the return can be just a slice without a pointer?

Reply:
Fixed in amended commit.